diff -up file-5.03/magic/Magdir/fonts.orig file-5.03/magic/Magdir/fonts
--- file-5.03/magic/Magdir/fonts.orig	2009-06-13 09:53:07.581775320 -0400
+++ file-5.03/magic/Magdir/fonts	2009-06-13 09:51:04.221775135 -0400
@@ -9,6 +9,8 @@
 0	string		%!PS-AdobeFont-1.	PostScript Type 1 font text
 >20	string		>\0			(%s)
 6	string		%!PS-AdobeFont-1.	PostScript Type 1 font program data
+6	string		%!FontType1	PostScript Type 1 font program data
+0	string		%!PS-Adobe-3.0\ Resource-Font	PostScript Type 1 font text
 
 # X11 font files in SNF (Server Natural Format) format
 0	belong		00000004		X11 SNF font data, MSB first
@@ -58,7 +60,7 @@
 0	string		ttcf		TrueType font collection data
 
 # Opentype font data from Avi Bercovich
-0	string		OTTO		OpenType font data 
+0	string		OTTO		OpenType font data
 
 # Gürkan Sengün <gurkan@linuks.mine.nu>, www.linuks.mine.nu 
 0	string		SplineFontDB:	Spline Font Database 
