--- file-4.19/src/magic.c.debian	2007-01-09 17:40:50.000000000 +0100
+++ file-4.19/src/magic.c	2007-01-09 17:42:29.000000000 +0100
@@ -238,7 +238,7 @@
 	int	fd = 0;
 	int	rv = -1;
 	unsigned char *buf;
-	struct stat	sb;
+	struct stat	sb, *st = &sb;
 	ssize_t nbytes = 0;	/* number of bytes read from a datafile */
 	int	ispipe = 0;
 
@@ -253,7 +253,7 @@
 	if (file_reset(ms) == -1)
 		goto done;
 
-	switch (file_fsmagic(ms, inname, &sb)) {
+	switch (file_fsmagic(ms, inname, st)) {
 	case -1:		/* error */
 		goto done;
 	case 0:			/* nothing found */
