diff -up file-4.26/magic/Magdir/perl.perl5 file-4.26/magic/Magdir/perl
--- file-4.26/magic/Magdir/perl.perl5	2008-12-04 09:44:05.000000000 +0100
+++ file-4.26/magic/Magdir/perl	2008-12-04 09:45:02.000000000 +0100
@@ -23,7 +23,6 @@
 # by Dmitry V. Levin and Alexey Tourbin
 # check the first line
 0	search/1	package
-0	regex		\^package[\ \t]+[A-Za-z_]
 >0	regex		\^package[\ \t]+[0-9A-Za-z_:]+\ *;	Perl5 module source text
 # not 'p', check other lines
 0	search/1	!p
