diff -up file-5.03/magic/Magdir/images.djvu file-5.03/magic/Magdir/images
--- file-5.03/magic/Magdir/images.djvu	2009-10-27 13:06:11.000000000 +0100
+++ file-5.03/magic/Magdir/images	2009-10-27 13:08:25.000000000 +0100
@@ -545,12 +545,12 @@
 # Reference: http://djvu.org/docs/DjVu3Spec.djvu
 # Submitted by: Stephane Loeuillet <stephane.loeuillet@tiscali.fr>
 # Modified by (1): Abel Cheung <abelcheung@gmail.com>
-0	string	AT&TFORM
+0	string	AT&TFORM	DjVu
 !:mime	image/vnd.djvu
->12	string	DJVM		DjVu multiple page document
->12	string	DJVU		DjVu image or single page document
->12	string	DJVI		DjVu shared document
->12	string	THUM		DjVu page thumbnails
+>12	string	DJVM		multiple page document
+>12	string	DJVU		image or single page document
+>12	string	DJVI		shared document
+>12	string	THUM		page thumbnails
 
 
 # From Marc Espie
