From 9f0601f3c03ffc561a2638c35eaf24308c7b3a72 Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos@zoulas.com>
Date: Fri, 2 Jan 2015 22:40:27 +0000
Subject: [PATCH] Explain recursion better and avoid double recursion in the
 masked entries

---
 magic/Magdir/jpeg | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/magic/Magdir/jpeg b/magic/Magdir/jpeg
index f32f58a..588073f 100644
--- a/magic/Magdir/jpeg
+++ b/magic/Magdir/jpeg
@@ -41,7 +41,8 @@
 # This uses recursion...
 0		name		jpeg_segment
 >0	beshort		0xFFFE
->>(2.S+2)	use			jpeg_segment
+# Recursion handled by FFE0
+#>>(2.S+2)	use			jpeg_segment
 >>2	pstring/HJ	x		\b, comment: "%s"
 
 >0	beshort		0xFFC0
@@ -70,6 +71,7 @@
 >>(2.S+2)	use			jpeg_segment
 
 >0	beshort		0xFFE1		
+# Recursion handled by FFE0
 #>>(2.S+2)	use			jpeg_segment
 >>4	string		Exif		\b, Exif Standard: [
 >>>10	indirect/r	x		
@@ -85,7 +87,8 @@
 # D9: End of image
 # Dn: Restart
 >0	beshort&0xFFD0	=0xFFD0
->>(2.S+2)	use			jpeg_segment
+>>0	beshort&0xFFE0	!0xFFE0
+>>>(2.S+2)	use			jpeg_segment
 
 #>0	beshort		x		unknown 0x%x
 #>>(2.S+2)	use			jpeg_segment
