diff -up file-4.26/magic/Magdir/msdos.devdrv file-4.26/magic/Magdir/msdos
--- file-4.26/magic/Magdir/msdos.devdrv	2008-10-07 16:00:29.000000000 +0200
+++ file-4.26/magic/Magdir/msdos	2008-10-07 16:02:01.000000000 +0200
@@ -293,22 +293,37 @@
 >10	string	>\x23			
 >>10	string	!\x2e			
 >>>17	string	<\x5B			
->>>>10	string	x			\b, name: %.8s
-#UDMA.SYS KEYB.SYS CMD640X2.SYS
+>>>>10	string	CMD640X2.SYS			\b, name: %.8s
+#UDMA.SYS
 >10	string	<\x41			
 >>12	string	>\x40			
 >>>10	string	!$			
->>>>12	string	x			\b, name: %.8s
-#BTCDROM.SYS ASPICD.SYS
+>>>>12	string	UDMA.SYS			\b, name: %.8s
+#CMD640X2.SYS
+>10	string	<\x41			
+>>12	string	>\x40			
+>>>10	string	!$			
+>>>>12	string	CMD640X2.SYS			\b, name: %.8s
+#KEYB.SYS
+>10	string	<\x41			
+>>12	string	>\x40			
+>>>10	string	!$			
+>>>>12	string	KEYB.SYS			\b, name: %.8s
+#BTCDROM.SYS
+>22	string	>\x40			
+>>22	string	<\x5B			
+>>>23	string	<\x5B			
+>>>>22	string	BTCDROM.SYS			\b, name: %.8s
+#ASPICD.SYS
 >22	string	>\x40			
 >>22	string	<\x5B			
 >>>23	string	<\x5B			
->>>>22	string	x			\b, name: %.8s
+>>>>22	string	ASPICD.SYS			\b, name: %.8s
 #ATAPICD.SYS
 >76	string	\0			
 >>77	string	>\x40			
 >>>77	string	<\x5B			
->>>>77	string	x			\b, name: %.8s
+>>>>77	string	ATAPICD.SYS			\b, name: %.8s
 #0	byte		0x8c		DOS executable (COM)
 # 0xeb conflicts with "sequent" magic
 #0	byte		0xeb		DOS executable (COM)
