--- file-4.17/magic/magic.mime.ppt	2006-06-26 08:57:42.000000000 +0200
+++ file-4.17/magic/magic.mime	2006-07-10 11:05:19.000000000 +0200
@@ -833,7 +833,7 @@
 
 0	string	PO^Q`			application/msword
 
-0	string	\320\317\021\340\241\261\032\341
+0	string	\320\317\021\340\241\261\032\341 application/msword
 >546	string	bjbj			application/msword
 >546	string	jbjb			application/msword
 
@@ -844,7 +844,6 @@
 2112	string	MSWordDoc			application/msword
 
 #0	string	\320\317\021\340\241\261\032\341	application/powerpoint
-0	string	\320\317\021\340\241\261\032\341	application/msword
 
 0       string  #\ PaCkAgE\ DaTaStReAm  application/x-svr4-package
 
