--- file-4.21/magic/Magdir/msdos.msoffice	2007-08-20 15:06:19.000000000 +0200
+++ file-4.21/magic/Magdir/msdos	2007-08-20 15:06:24.000000000 +0200
@@ -624,5 +624,5 @@
 0	lelong	0x02468ace			Bochs Sparse disk image
 
 # from http://filext.com by Derek M Jones <derek@knosof.co.uk>
-0	string	\xD0\xCF\x11\xE0\xA1\xB1\x1A\xE1\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x3E\x00\x03\x00\xFE\xFF	Microsoft Installer
+#0	string	\xD0\xCF\x11\xE0\xA1\xB1\x1A\xE1\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x3E\x00\x03\x00\xFE\xFF	Microsoft Installer
 0	string	\320\317\021\340\241\261\032\341	Microsoft Office Document
--- file-4.21/magic/magic.mime.msoffice	2007-08-17 11:17:19.000000000 +0200
+++ file-4.21/magic/magic.mime	2007-08-17 15:01:58.000000000 +0200
@@ -851,6 +851,7 @@
 0	string	\320\317\021\340\241\261\032\341
 >546	string	bjbj			application/msword
 >546	string	jbjb			application/msword
+>546    default x               application/msword
 
 512	string	R\0o\0o\0t\0\ \0E\0n\0t\0r\0y	application/msword
 
@@ -859,7 +860,6 @@
 2112	string	MSWordDoc			application/msword
 
 #0	string	\320\317\021\340\241\261\032\341	application/powerpoint
-0	string	\320\317\021\340\241\261\032\341	application/msword
 
 0       string  #\ PaCkAgE\ DaTaStReAm  application/x-svr4-package
 
