diff -up file-5.00/magic/Magdir/fonts.ttcpfa file-5.00/magic/Magdir/fonts
--- file-5.00/magic/Magdir/fonts.ttcpfa	2009-03-23 11:44:36.000000000 +0100
+++ file-5.00/magic/Magdir/fonts	2009-03-23 11:52:48.000000000 +0100
@@ -9,6 +9,7 @@
 0	search/1	%!PS-AdobeFont-1.	PostScript Type 1 font text
 >20	search/1	>\0			(%s)
 6	string		%!PS-AdobeFont-1.	PostScript Type 1 font program data
+0	string		%!PS-AdobeFont-1.	PostScript Type 1 font text
 
 # X11 font files in SNF (Server Natural Format) format
 0	belong		00000004		X11 SNF font data, MSB first
@@ -55,6 +56,8 @@
 0	string		\007\001\001\000Copyright\ (c)\ 199	Adobe Multiple Master font
 0	string		\012\001\001\000Copyright\ (c)\ 199	Adobe Multiple Master font
 
+0	string		ttcf		TrueType font collection data
+
 # Opentype font data from Avi Bercovich
 0	string		OTTO		OpenType font data 
 
