diff -up file-4.24/magic/Makefile.am.selinux file-4.24/magic/Makefile.am
--- file-4.24/magic/Makefile.am.selinux	2008-03-22 22:41:00.000000000 +0100
+++ file-4.24/magic/Makefile.am	2008-05-06 13:32:36.000000000 +0200
@@ -149,6 +149,7 @@ $(MAGIC_FRAGMENT_DIR)/pgp \
 $(MAGIC_FRAGMENT_DIR)/pkgadd \
 $(MAGIC_FRAGMENT_DIR)/plan9 \
 $(MAGIC_FRAGMENT_DIR)/plus5 \
+$(MAGIC_FRAGMENT_DIR)/policy \
 $(MAGIC_FRAGMENT_DIR)/printer \
 $(MAGIC_FRAGMENT_DIR)/project \
 $(MAGIC_FRAGMENT_DIR)/psdbms \
diff -up /dev/null file-4.24/magic/Magdir/policy
--- /dev/null	2008-04-14 09:39:21.445009731 +0200
+++ file-4.24/magic/Magdir/policy	2008-05-06 13:29:35.000000000 +0200
@@ -0,0 +1,6 @@
+# SE Linux policy database
+0       lelong  0xf97cff8c      SE Linux policy
+>16     lelong  x               v%d
+>20     lelong  1 		MLS
+>24     lelong  x       	%d symbols
+>28     lelong  x       	%d ocons
