--- file-4.19/magic/Magdir/archive.ooffice	2007-01-25 11:18:43.000000000 +0100
+++ file-4.19/magic/Magdir/archive	2007-01-25 11:19:08.000000000 +0100
@@ -536,8 +536,6 @@
 >4	byte		0x0b		Zip archive data, at least v1.1 to extract
 >4	byte		0x14
 >>30	ubelong		!0x6d696d65	Zip archive data, at least v2.0 to extract
->0x161	string		WINZIP          Zip archive data, WinZIP self-extracting
-
 
 # OpenOffice.org / KOffice / StarOffice documents
 # From: Abel Cheung <abel@oaka.org>
