--- file-4.16/magic/Localstuff.xen	2003-03-23 05:17:27.000000000 +0100
+++ file-4.16/magic/Localstuff	2006-02-02 14:17:25.000000000 +0100
@@ -5,3 +5,9 @@
 # $Id: Localstuff,v 1.4 2003/03/23 04:17:27 christos Exp $
 # Add any locally observed files here.  Remember:
 # text if readable, executable if runnable binary, data if unreadable.
+
+# File magic for Xen, the virtual machine monitor for x86
+0	string		LinuxGuestRecord	Xen saved domain
+#>2	regex		\(name\ [^)]*\)		%s
+>20	search/256	(name			(name
+>>&1	string		x			%s...)
