diff -up file-5.03/magic/Magdir/linux.ppcswap file-5.03/magic/Magdir/linux
--- file-5.03/magic/Magdir/linux.ppcswap	2009-10-21 15:48:42.000000000 +0200
+++ file-5.03/magic/Magdir/linux	2009-10-21 15:50:09.000000000 +0200
@@ -61,6 +61,9 @@
 >0x404	long		x		size %d pages
 >>4086	string		SWAPSPACE2	
 >>>1052	string		>\0		Label %s
+# From Daniel Novotny <dnovotny@redhat.com>
+# swap file for PowerPC
+65526   string          SWAPSPACE2      Linux/ppc swap file
 # ECOFF magic for OSF/1 and Linux (only tested under Linux though)
 #
 #	from Erik Troan (ewt@redhat.com) examining od dumps, so this
