diff -up file-4.24/magic/Makefile.in.berkeley file-4.24/magic/Makefile.in
--- file-4.24/magic/Makefile.in.berkeley	2008-05-06 13:49:07.000000000 +0200
+++ file-4.24/magic/Makefile.in	2008-05-06 13:50:41.000000000 +0200
@@ -204,10 +204,10 @@ $(MAGIC_FRAGMENT_DIR)/communications \
 $(MAGIC_FRAGMENT_DIR)/compress \
 $(MAGIC_FRAGMENT_DIR)/console \
 $(MAGIC_FRAGMENT_DIR)/convex \
-$(MAGIC_FRAGMENT_DIR)/cracklib \
 $(MAGIC_FRAGMENT_DIR)/ctags \
 $(MAGIC_FRAGMENT_DIR)/dact \
 $(MAGIC_FRAGMENT_DIR)/database \
+$(MAGIC_FRAGMENT_DIR)/cracklib \
 $(MAGIC_FRAGMENT_DIR)/diamond \
 $(MAGIC_FRAGMENT_DIR)/diff \
 $(MAGIC_FRAGMENT_DIR)/digital \
