--- file-4.15/magic/Makefile.in.berkeley	2005-10-03 11:26:31.000000000 +0200
+++ file-4.15/magic/Makefile.in	2005-10-03 11:27:28.000000000 +0200
@@ -200,7 +200,6 @@
 Magdir/citrus \
 Magdir/claris \
 Magdir/clipper \
-Magdir/cracklib \
 Magdir/spec \
 Magdir/commands \
 Magdir/communications \
@@ -210,6 +209,7 @@
 Magdir/ctags \
 Magdir/dact \
 Magdir/database \
+Magdir/cracklib \
 Magdir/diamond \
 Magdir/diff \
 Magdir/digital \
