diff -up file-4.23/magic/Makefile.in.fsdump file-4.23/magic/Makefile.in
--- file-4.23/magic/Makefile.in.fsdump	2008-01-28 11:23:20.000000000 +0100
+++ file-4.23/magic/Makefile.in	2008-01-28 11:23:35.000000000 +0100
@@ -215,10 +215,10 @@ Magdir/diamond \
 Magdir/diff \
 Magdir/digital \
 Magdir/dolby \
-Magdir/dump \
 Magdir/editors \
 Magdir/efi \
 Magdir/elf \
+Magdir/dump \
 Magdir/encore \
 Magdir/epoc \
 Magdir/esri \
