diff -up file-5.03/magic/Makefile.am.multilib file-5.03/magic/Makefile.am
--- file-5.03/magic/Makefile.am.multilib	2009-11-30 11:38:18.000000000 +0100
+++ file-5.03/magic/Makefile.am	2009-11-30 11:47:23.000000000 +0100
@@ -234,5 +234,7 @@ FILE_COMPILE_DEP = $(FILE_COMPILE)
 endif
 
 ${MAGIC}: $(EXTRA_DIST) $(FILE_COMPILE_DEP)
-	$(FILE_COMPILE) -C -m $(MAGIC_FRAGMENT_DIR)
-	@mv $(MAGIC_FRAGMENT_BASE).mgc $@
+	cat $(MAGIC_FRAGMENT_DIR)/* > all-magic
+	$(FILE_COMPILE) -C -m all-magic
+	rm all-magic
+	@mv all-magic.mgc $@
diff -up file-5.03/magic/Makefile.in.multilib file-5.03/magic/Makefile.in
--- file-5.03/magic/Makefile.in.multilib	2009-11-30 11:38:34.000000000 +0100
+++ file-5.03/magic/Makefile.in	2009-11-30 11:48:28.000000000 +0100
@@ -584,8 +584,10 @@ uninstall-am: uninstall-pkgdataDATA
 
 
 ${MAGIC}: $(EXTRA_DIST) $(FILE_COMPILE_DEP)
-	$(FILE_COMPILE) -C -m $(MAGIC_FRAGMENT_DIR)
-	@mv $(MAGIC_FRAGMENT_BASE).mgc $@
+	cat $(MAGIC_FRAGMENT_DIR)/* > all-magic
+	$(FILE_COMPILE) -C -m all-magic
+	rm all-magic
+	@mv all-magic.mgc $@
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
